harness/determined/common/experimental/project.py

```diff
@@ -0,0 +1,214 @@
+import itertools
+from typing import Any, Dict, List, Optional
+
+from determined.common import api
+from determined.common.api import bindings
+from determined.common.experimental import experiment
+
+
+class Project:
+    """A class representing a Project object.
+
+    Attributes:
+        id: (int) The ID of the project.
+        archived: (Mutable, bool) True if experiment is archived, else false.
+        description: (Mutable, str) The description of the project.
+        n_active_experiments: (int) The number of active experiments in the project.
+        n_experiments: (Mutable, int) The number of experiments in the project.
+        name: (Mutable, str) Human-friendly name of the project.
+        notes: (Mutable, List[Dict[str,str]) Notes about the project. As determined upstream,
+            each note is a dict with exactly the keys "name" and "contents".
+        username: (Mutable, str) The username of the project owner.
+        workspace_id: (int) The ID of the workspace this project belongs to.
+    """
+
+    def __init__(
+        self,
+        session: api.Session,
+        project_id: int,
+    ):
+        """Create a Project object.
+
+        Arguments:
+            session: The session to use for API calls.
+            project_id: ID of the project.
+        """
+        self._session = session
+        self.id = project_id
+
+        # These properties may be mutable and will be set by _hydrate()
+        self.archived: Optional[bool] = None
+        self.description: Optional[str] = None
+        self.n_active_experiments: Optional[int] = None
+        self.n_experiments: Optional[int] = None
+        self.name: Optional[str] = None
+        self.notes: Optional[List[Dict[str, str]]] = None
+        self.workspace_id: Optional[int] = None
+        self.username: Optional[str] = None
+
+    @classmethod
+    def _from_bindings(
+        cls, project_bindings: bindings.v1Project, session: api.Session
+    ) -> "Project":
+        project = cls(session, project_id=project_bindings.id)
+        project._hydrate(project_bindings)
+        return project
+
+    def _hydrate(self, project_bindings: bindings.v1Project) -> None:
+        """Set this object's mutable attributes from those in a bindings object."""
```

Comment:
"""Set this object's mutable attributes to those in the bindings object."""

Reply:
"From" is slightly more accurate, here, since there's a small transformation in the bindings object's attributes that happens before setting the mutable attributes. Even without that transformation, "from" feels to me like it captures execution better. Could you help me understand your thinking?

Reply:
Let's keep "from" if it is more accurate. It sounded awkward when I read it out loud.